src/nvim/main.c

```diff
@@ -1444,6 +1444,19 @@ static void command_line_scan(mparm_T *parmp)
       ga_grow(&global_alist.al_ga, 1);
       char *p = xstrdup(argv[0]);
 
+      // On Windows expand "~\" or "~/" prefix in file names to profile directory.
+#ifdef MSWIN
+      if (*p == '~' && (p[1] == '\\' || p[1] == '/')) {
+        char *profile_dir = vim_getenv("HOME");
```

Comment:
Does this happen before `init_homedir`, or why not use `os_homedir` ? $HOME is not always available Windows.

Also needs test coverage.

But I'm confused about why this is needed, because https://github.com/neovim/neovim/issues/23901#issuecomment-2033479836 indicated that powershell would provide this feature without every program needing to implement it manually.